### DISCUSSION TITLE
Obtain injected chain data on the wallet even if the wallet is not connected to the app

### DISCUSSION THREAD

**Discussion by berkaycirak** [rainbow-me/rainbowkit#1916] [Q&A]
Hi, I am developing a dapp. And curious about to get chain data even if the user hasn't connected yet to the app. For example, in my metamask, I select Avalanche Network but hasn't connected the app yet.
Normally, when I connect the app, I can listen the chain by `useChainId` hook. Without connecting, is there a way to do that?

**Comment by 0xmkh** [comment]
@berkaycirak Yes you can use `useChainid` and `useSwitchChain` hook from wagmi. This will allow you to read chain id without having your wallet connected.

If you're also looking to get the name and other info you can use `useConfig` and pull `chains` to see what current chain you want to display based on what your `useChainId` hook returns.
